### ISSUE TITLE
Prevent messages containing a known Jira instance URL from parsing this URL

### ISSUE THREAD

**Issue by d-gubert** [RocketChat/Apps.Atlassian.Jira#7]
Whenever a user puts the URL of a Jira issue, for instance, in the message, Rocket.Chat is going to show a preview similar to this: 
![image](https://user-images.githubusercontent.com/1810309/48021686-52d9ab00-e120-11e8-8697-1cce62bab768.png)

That is because Jira redirects not logged in access to the login page.

The app should prevent messages containing such URLs from showing their preview, additionally to the bevaior of intercepting them and showing the details of the issue there, if present.

**NOTE:** At the time of creation of this issue, the Rocket.Chat Apps Engine doesn't support this action. Solving this issue depends on the engine alteration

**Comment by graywolf336** [comment]
Actually, you could implement the pre-message sent modification interface and if it contains a known link then add `parseUrls: false` to not display that. As a temporary option at least

**Comment by wreiske** [comment]
IPreMessageSentPrevent 
https://github.com/cardoso/Rocket.Chat.ContentFilter/blob/master/ContentFilterApp.ts#L23

Example of how to do it

**Comment by d-gubert** [comment]
@graywolf336 the engine doesn't seem to pass the `parseUrl` attribute from Rocket.Chat to App and vice-versa, right?

**Comment by d-gubert** [comment]
> IPreMessageSentPrevent
> https://github.com/cardoso/Rocket.Chat.ContentFilter/blob/master/ContentFilterApp.ts#L23
> 
> Example of how to do it

We don't want to prevent the message from being sent, we just want to prevent it from previewing the URL as an attachment. This would require the `parseUrl` attribute as @graywolf336 said, but it seems we don't have access to it from the app

**Comment by wreiske** [comment]
Could you prevent it from being sent, parse it and grab the details from it, and then post the message from the app as the user?

**Comment by d-gubert** [comment]
I think Rocket.Chat still parses URLs from messages sent from apps

**Comment by wreiske** [comment]
I don't think it does, I am doing it in my WebEx app and it doesn't parse the URL when posted from the app

**Comment by d-gubert** [comment]
Now that's interesting. Gonna test this out tomorrow. Still I think the engine should support this, as it feels more natural than pretending to be an user, imo

**Comment by graywolf336** [comment]
Yeah, looks like the `parseUrls` property on messages was overlooked.

**Comment by wreiske** [comment]
@graywolf336 should this be as simple as adding parseUrls to https://github.com/RocketChat/Rocket.Chat.Apps-engine/blob/d3521eaec7979dbf3c31e0ca050c611f6f235840/src/definition/messages/IMessage.ts

**Comment by graywolf336** [comment]
Yes, it would be a very simple addition :) feel free to submit a pull request to add it :)
